src/System.Globalization.Calendars/tests/GregorianCalendar/GregorianCalendarTests.Utilities.cs

```diff
@@ -6,7 +6,21 @@ namespace System.Globalization.Tests
 {
     public class GregorianCalendarTestUtilities
     {
-        private static readonly RandomDataGenerator s_randomDataGenerator = new RandomDataGenerator();
+        [ThreadStatic]
+        private static RandomDataGenerator s_randomDataGenerator;
+
+        private static RandomDataGenerator RandomDataGenerator
+        {
+            get
+            {
+                if (s_randomDataGenerator == null)
+                {
+                    s_randomDataGenerator = new RandomDataGenerator();
+                }
+                return s_randomDataGenerator;
+            }
+        }
```

Comment:
Nit: it'd be good to also name the property something other than the same name as the class... otherwise the call sites look like they're static method calls to methods on RandomDataGenerator rather than instance method calls on the object returned from the prop